app/src/main/java/org/oppia/android/app/settings/profile/SnackbarController.kt

```diff
@@ -0,0 +1,43 @@
+package org.oppia.android.app.settings.profile
+
+import androidx.annotation.StringRes
+import org.oppia.android.util.data.DataProvider
+import org.oppia.android.util.data.DataProviders
+import java.util.LinkedList
+import java.util.Queue
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class SnackbarController @Inject constructor(private val dataProviders: DataProviders) {
+
+  private val snackbarRequestQueue: Queue<SnackbarRequest.ShowSnackbar> = LinkedList()
+
+  fun getCurrentSnackbar(): DataProvider<SnackbarRequest> {
+    val currentRequest = snackbarRequestQueue.peek()
+    return dataProviders.createInMemoryDataProvider(SnackbarRequest::class.java) {
+      return@createInMemoryDataProvider currentRequest
+        ?: SnackbarRequest.ShowNothing
+    }
+  }
+
+  fun enqueueSnackbar(request: SnackbarRequest.ShowSnackbar) {
+    snackbarRequestQueue.add(request)
+  }
+
+  fun dismissCurrentSnackbar() {
+    snackbarRequestQueue.remove()
```

Comment:
Both of these will need to notify that the data provider changed in order to the mechanism to show/hide the snackbar to work correctly. See: https://github.com/oppia/oppia-android/blob/7e252747473e02463be21e11c75b36b048fd4690/domain/src/main/java/org/oppia/android/domain/locale/LocaleController.kt#L137

Reply:
corrected